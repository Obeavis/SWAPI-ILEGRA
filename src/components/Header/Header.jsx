import React from "react";
import { withRouter } from "react-router-dom";
import { Navbar } from "react-bootstrap";
import logo from "../../assets/images/star-wars-logo.png";
import "./Header.scss";

const Header = (props) => (
	<Navbar className="justify-content-center bg-primary header fixed-top">
		<Navbar.Brand>
			<img
				alt=""
				src={logo}
				width="150"
				height="100"
				className="img-fluid"
				onClick={() => props.history.push(`/main`)}
			/>
		</Navbar.Brand>

	</Navbar>
);

export default withRouter(Header);