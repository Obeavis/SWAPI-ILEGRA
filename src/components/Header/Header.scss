.header {
    box-shadow: 0px 13px 35px -20px rgba(61, 61, 61, 1);
    
    .navbar-brand {
        margin: 0 !important;
        cursor: pointer;
    }
}