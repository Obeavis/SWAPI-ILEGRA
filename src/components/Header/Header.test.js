import React from "react";
import { MemoryRouter  } from 'react-router-dom';
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import Header from "./Header";

configure({ adapter: new Adapter() });

describe("Header Component", () => {
    
    const component = (

        <MemoryRouter initialEntries={['/films']}>
                <Header />
        </MemoryRouter>
    )

    it("matches the snapshot", () => {
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it("Renders the image", () => {
        const wrapper = mount(component);
        const image = wrapper.find('img').prop("src");

        expect(image).toEqual('star-wars-logo.png');

    });

    it("Route to Main when click the image", () => {
        const wrapper = mount(component);
        const pathnameBeforeClick =  wrapper.find('Router').props().history.location.pathname;
        expect(pathnameBeforeClick).toEqual('/films');

        wrapper.find('img').simulate('click');

        const pathnameAfterClick =  wrapper.find('Router').props().history.location.pathname;
        expect(pathnameAfterClick).toEqual('/main');
    });

})