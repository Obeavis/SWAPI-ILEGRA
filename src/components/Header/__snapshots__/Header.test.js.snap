// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches the snapshot 1`] = `
<nav
  className="justify-content-center bg-primary header fixed-top navbar navbar-expand navbar-light"
>
  <span
    className="navbar-brand"
  >
    <img
      alt=""
      className="img-fluid"
      height="100"
      onClick={[Function]}
      src="star-wars-logo.png"
      width="150"
    />
  </span>
</nav>
`;
